### PR TITLE
Rac 4090/ha deployment

### PR DESCRIPTION
**Pull request by johren**
Configures the control network, DHCP, /etc/hosts, and hostname on all nodes in the cluster.

Checks the corosync and pacemaker status to verify that we have created a happy cluster.

Had to modify fit_common.remote_shell and fit_common.scp_file_to_ora to pass in a VM number.
Defaults to node 1.

Tested by running the following with 3 and 4 node clusters.
```
python ./run_tests.py -test deploy/os_ova_install.py -stack 4 -template http://mirrors.hwimo.lab.emc.com/mirrors/ova/ha-ubuntu-1604.ova -config dellemc-test/config-mn -numvms 4

python ./run_tests.py -test deploy/rackhd_ha_install.py -config dellemc-test/config-mn -stack 4 -numvms 4
```

### CODE REVIEW

File: test/deploy/rackhd_ha_install.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            os.remove('dhcpd.conf')
+
+    def create_hosts(self):
+        hosts_conf = open('hosts-conf', 'w')
+        for vmnum in range(1, numvms + 1):
+            line = '172.31.128.{}\tnode{}\n'.format(vmnum, vmnum)
+            hosts_conf.write(line)
+        hosts_conf.close()
+
+    def test04_install_hosts_config(self):
```
Comment (hohene) [inline test/deploy/rackhd_ha_install.py]: test04 - here and below, looks like copy/paste issue

File: test/deploy/rackhd_ha_install.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        os.remove('hosts-conf')
+
+    def test04_install_hostname_config(self):
+        for vmnum in range(1, numvms + 1):
+            command = "echo node{} > /etc/hostname".format(vmnum)
+            self.assertEqual(fit_common.remote_shell(command, vmnum=vmnum)['exitcode'], 0, "Hostname Config failure.")
+
+
+if __name__ == '__main__':
+    fit_common.unittest.main()
```
Comment (hohene) [inline test/deploy/rackhd_ha_install.py]: can get rid of fit_common here  - ie. import unittest above and change to unittest.main

File: test/deploy/rackhd_ha_install.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            self.assertEqual(fit_common.remote_shell('cp control.cfg /etc/network/interfaces.d/', vmnum=vmnum)['exitcode'],
+                             0, "Control network config failure.")
+            os.remove('control.cfg')
+            # startup NIC
+            fit_common.remote_shell('ip addr add ' + control_ip + '/22 dev ' + ifslist[1], vmnum=vmnum)
+            fit_common.remote_shell('ip link set ' + ifslist[1] + ' up', vmnum=vmnum)
+            self.assertEqual(fit_common.remote_shell('ping -c 1 -w 5 ' + control_ip, vmnum=vmnum)['exitcode'],
+                             0, 'Control NIC failure.')
+
+    def test02_install_dhcp_config(self):
```
Comment (hohene) [inline test/deploy/rackhd_ha_install.py]: Suggestions: For a deploy script, you could use nosedep and @depends if you want some of the tests to be skipped if the previous one failed.

Comment (johren) [inline test/deploy/rackhd_ha_install.py]: Great suggestion!  Thanks.

File: test/deploy/rackhd_ha_install.py (hunk 1 above)
Comment (johren) [inline test/deploy/rackhd_ha_install.py]: Will fix

File: test/deploy/rackhd_ha_install.py (hunk 2 above)
Comment (johren) [inline test/deploy/rackhd_ha_install.py]: Cut-n-paste.  Will fix.

**Comment by johren** [comment]
test this please

**Comment by yyscamper** [comment]
@johren : please squash your commit history.

File: test/deploy/rackhd_ha_install.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+from jinja2 import Environment, FileSystemLoader
+import os
+import sys
+import subprocess
+import unittest
+import time
+from nosedep import depends
+
+# set path to common libraries
+sys.path.append(subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test/common")
```
Comment (hohene) [inline test/deploy/rackhd_ha_install.py]: Update this script to use fit_path and gets rid of this sys.path.append line.  (see the test/template file.)

File: test/deploy/rackhd_ha_install.py [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+            addrsplit = address.split('.')
+            return("{}.{}.{}".format(addrsplit[0], addrsplit[1], addrsplit[2]))
+        return None
+
+    def create_corosync_config(self):
+        template = self.get_template_file("corosync_config")
+        rendered = template.render(node_list=self.create_node_list())
+        return rendered
+
+    def test03_install_hosts_config(self):
```
Comment (hohene) [inline test/deploy/rackhd_ha_install.py]: do you want this to depend on prior test results?   if so, add @depends(after=xxxx)

File: test/deploy/rackhd_ha_install.py [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+                             0, "Hosts Config failure.")
+            # Add the new entries
+            self.assertEqual(fit_common.remote_shell('cat hosts-conf >> hosts', vmnum=vmnum)['exitcode'],
+                             0, "Hosts Config failure.")
+            # Move the new file into place
+            self.assertEqual(fit_common.remote_shell('mv hosts /etc/hosts', vmnum=vmnum)['exitcode'],
+                             0, "Hosts Config failure.")
+        os.remove('hosts-conf')
+
+    def test04_install_hostname_config(self):
```
Comment (hohene) [inline test/deploy/rackhd_ha_install.py]: same comment - add the @depend(after=xxx) if you want the test to be skipped.  Same with subsequent test05, 06, 07.

File: test/deploy/rackhd_ha_install.py (hunk 4 above)
Comment (johren) [inline test/deploy/rackhd_ha_install.py]: Okay if I fix that in a quick follow-on PR?  I currently don't need to rebase and it would be easier to do a separate PR.

Comment (hohene) [inline test/deploy/rackhd_ha_install.py]: yep, it should still work as is.

**Comment by johren** [comment]
Squashed and rebased and then realized that part of the rebase was changing scp_file_to_ora to scp_file_to_host.  I have incorporated that change into rackhd_ha_install.py and I am currently testing that change.

**Comment by johren** [comment]
@hohene @gpaulos @larry-dean @patelb10

**Comment by johren** [comment]
test this please
